apps/web/components/blog/blog-page.tsx

```diff
@@ -0,0 +1,164 @@
+"use client";
+
+import type {
+  BlogCategory,
+  BlogPostListItem,
+  BlogSortOptions,
+} from "@/lib/blog/types";
+import { useMemo, useState } from "react";
+import { BlogFilters } from "./list/blog-filters";
+import { BlogSearch } from "./list/blog-search";
+import { BlogSort } from "./list/blog-sort";
+import { BlogCard } from "./shared/blog-card";
+import { FeaturedPostCard } from "./shared/featured-post-card";
+
+interface BlogPageProps {
+  posts: BlogPostListItem[];
+  featuredPosts?: BlogPostListItem[];
+}
+
+const GRADIENTS = [
+  "bg-gradient-to-br from-emerald-300 via-cyan-300 to-blue-400",
+  "bg-gradient-to-br from-pink-300 via-rose-300 to-emerald-300",
+  "bg-gradient-to-br from-blue-400 via-indigo-300 to-pink-300",
+  "bg-gradient-to-br from-yellow-300 via-lime-300 to-emerald-300",
+  "bg-gradient-to-br from-teal-400 via-cyan-400 to-blue-400",
+  "bg-gradient-to-br from-emerald-400 via-teal-300 to-pink-300",
+];
+
+export function BlogPage({ posts, featuredPosts = [] }: BlogPageProps) {
+  const [searchQuery, setSearchQuery] = useState("");
+  const [selectedCategory, setSelectedCategory] = useState<
+    BlogCategory | "all"
+  >("all");
+  const [sortOptions, setSortOptions] = useState<BlogSortOptions>({
+    field: "date",
+    order: "desc",
+  });
+
+  // Calculate post counts per category
+  const postCounts = useMemo(() => {
+    const counts: Record<BlogCategory | "all", number> = {
+      all: posts.length,
+      new: 0,
+      feature: 0,
+      "bug fix": 0,
+      update: 0,
+      event: 0,
+      tutorial: 0,
+      announcement: 0,
+    };
+
+    posts.forEach((post) => {
+      counts[post.category]++;
+    });
+
+    return counts;
+  }, [posts]);
+
+  // Filter and sort posts
+  const filteredPosts = useMemo(() => {
+    let filtered = [...posts];
+
+    // Apply search filter
+    if (searchQuery) {
+      const query = searchQuery.toLowerCase();
```

Comment:
at the very least use `.toLocaleLowerCase()` - consider `toLowerCase()` and `toUpperCase()` to be deprecated.